Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoUtils.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/std/string/string.h>
+#include <AzToolsFramework/Prefab/PrefabDomTypes.h>
+
+namespace AzToolsFramework
+{
+    namespace Prefab
+    {
+        namespace PrefabUndoUtils
+        {
+            void GenerateAddEntityPatch(
+                PrefabDom& patch,
+                PrefabDom& newEntityDom,
+                const AZStd::string& newEntityAliasPath);
+
+            void GenerateRemoveEntityPatch(
+                PrefabDom& patch,
+                const AZStd::string& targetEntityAliasPath);
+
+            void GenerateUpdateEntityPatch(
```

Comment:
nit: It might be good to rename this 'Generate' functions to be 'Append' functions since we are appending patches to the passed dom. Example: AppendUpdateEntityPatches()

Reply:
`GenerateUpdateEntityPatch` actually creates an array of changes in given patch so I will keep using this name, but we can change the other two functions' name to `AppendAddEntityPatch` and `AppendRemoveEntityPatch`.